### PR TITLE
[MDEP-776] Warn when multiple dependencies have the same file name

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java [hunk 1]
```diff
@@ -111,6 +109,21 @@ protected void doExecute() throws MojoExecutionException {
...
+            for (Artifact artifactItem : artifacts) {
+                String destFileName = DependencyUtil.getFormattedFileName(
+                        artifactItem, stripVersion, prependGroupId, useBaseVersion, stripClassifier);
+                int numCopies = copies.getOrDefault(destFileName, 0);
+                copies.put(destFileName, numCopies + 1);
+            }
+            for (Map.Entry<String, Integer> entry : copies.entrySet()) {
+                if (entry.getValue() > 1) {
+                    getLog().warn("Multiple files with the name " + entry.getKey() + " in the dependency tree.");
+                    getLog().warn(
```
Comment (Bukama) [inline src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java:122]: Is this new line on purpose?

**Review by Bukama — APPROVED** [review]
I like that you also updated so much documentation / Javadoc!
I only have one small, little question, but that only might be a personal nuance. 

So non-binding approval by me.

File: src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java (hunk 1 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java:122]: It's weird but that's what spotless produces

File: src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java [hunk 2]
```diff
@@ -111,6 +109,21 @@ protected void doExecute() throws MojoExecutionException {
...
+                                    "Not all jars will be available. Consider using prependGroupId, useSubDirectoryPerArtifact, or useRepositoryLayout.");
```
Comment (michael-o) [inline src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java:123]: JARs

**Review by michael-o — DISMISSED** [review]

File: src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java (hunk 2 above)
Comment (elharo) [reply src/main/java/org/apache/maven/plugins/dependency/fromDependencies/CopyDependenciesMojo.java]: done

**Comment by elharo** [comment]
ping

File: src/it/projects/copy-dependencies-with-conflict/pom.xml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.apache</groupId>
+    <artifactId>apache</artifactId>
+    <version>5</version>
```
Comment (slawekjaranowski) [inline src/it/projects/copy-dependencies-with-conflict/pom.xml:29]: do we need a parent for test ... if yes - should be something newer

**Review by slawekjaranowski — APPROVED** [review]

File: src/it/projects/copy-dependencies-with-conflict/pom.xml (hunk 3 above)
Comment (elharo) [reply src/it/projects/copy-dependencies-with-conflict/pom.xml]: not really. Removed.

**Comment by jira-importer** [comment]
Resolve #1283

**Pull request merged**